src/dhtproto/client/mixins/NeoSupport.d

```diff
@@ -901,6 +937,145 @@ template NeoSupport ( )
 
         /***********************************************************************
 
+            Assigns an Exists request and blocks the current Task until the
+            request is completed. See
+            $(LINK2 dhtproto/client/request/Exists.html, dhtproto.client.request.Exists)
+            for detailed documentation.
+
+            Params:
+                channel = name of the channel to check in
+                key = hash of the record to check for
+                user_notifier = notifier delegate
+
+        ***********************************************************************/
+
+        public void exists ( cstring channel, hash_t key,
+            Neo.Exists.Notifier user_notifier )
+        in
+        {
+            assert(user_notifier !is null);
+        }
+        body
+        {
+            auto task = Task.getThis();
+            assert(task, "This method may only be called from inside a Task");
+
+            bool finished;
+
+            void notifier ( Neo.Exists.Notification info, Neo.Exists.Args args )
+            {
+                user_notifier(info, args);
+
+                // All Exists notifications indicate that the request has
+                // finished (no need to switch)
+                finished = true;
+                if ( task.suspended )
+                    task.resume();
+            }
+
+            this.outer.neo.exists(channel, key, &notifier);
+            if ( !finished ) // if request not completed, suspend
+                task.suspend();
+            assert(finished);
+        }
+
+        /***********************************************************************
+
+            Struct returned after an Exists request has finished.
+
+        ***********************************************************************/
+
+        private static struct ExistsResult
```

Comment:
This could be an `enum ExistsResult { Error, Exists, NoRecord }`.

Reply:
Yeah, I made it a struct for consistency with the other task-blocking methods.